internal/verifier/migration_verifier_test.go

```diff
@@ -1545,6 +1553,82 @@ func (suite *IntegrationTestSuite) TestVerifierWithFilter() {
 	<-checkDoneChan
 }
 
+func (suite *IntegrationTestSuite) waitForRecheckDocs(verifier *Verifier) {
+	suite.Eventually(func() bool {
+		cursor, err := suite.metaMongoClient.Database(verifier.metaDBName).Collection(recheckQueue).Find(suite.Context(), bson.D{})
+		var docs []bson.D
+		suite.Require().NoError(err)
+		suite.Require().NoError(cursor.All(suite.Context(), &docs))
+		return len(docs) > 0
+	}, 1*time.Minute, 100*time.Millisecond)
+}
+
+func (suite *IntegrationTestSuite) TestChangesOnDstBeforeSrc() {
+	zerolog.SetGlobalLevel(zerolog.TraceLevel)
+	defer zerolog.SetGlobalLevel(zerolog.DebugLevel)
+
+	ctx := suite.Context()
+
+	collName := "mycoll"
+
+	srcDB := suite.srcMongoClient.Database(suite.DBNameForTest())
+	dstDB := suite.dstMongoClient.Database(suite.DBNameForTest())
+	suite.Require().NoError(srcDB.CreateCollection(ctx, collName))
+	suite.Require().NoError(dstDB.CreateCollection(ctx, collName))
+
+	verifier := suite.BuildVerifier()
+	runner := RunVerifierCheck(ctx, suite.T(), verifier)
+
+	// Dry run generation 0 to make sure change stream reader is started.
+	suite.Require().NoError(runner.AwaitGenerationEnd())
+
+	suite.Require().NoError(runner.StartNextGeneration())
```

Comment:
Should the generation start after the inserts? It seems like that’d be a bit less race-prone.

Reply:
Looks like it's more likely to race if inserts happen after generation start because StartNextGeneration isn't blocked on the next generation. I've reverted the changes for this test so that change events occur in the previous generation and tasks should appear in the next generation.